Error extracting flashlight executable because another one is already running

2014-11-17 23:02:04.529 UTC
{ "reporterId" : "f1647a0ae93f0a4d1fa008647cf35603", "message" : "Uncaught throwable", "fullMessage" : "Uncaught throwable", "occurredAt" : 1416265286624, "locationInfo" : "org.littleshoot.proxy.impl.DefaultHttpProxyServer$ServerGroup$1.uncaughtException(DefaultHttpProxyServer.java:440)", "stackTrace" : "java.lang.Error: Could not extract jar file from:pt/flashlight/flashlight.exe\r\n\tat org.lantern.proxy.pt.BasePluggableTransport.<init>(BasePluggableTransport.java:67)\r\n\tat org.lantern.proxy.pt.Flashlight.<init>(Flashlight.java:57)\r\n\tat org.lantern.proxy.pt.FlashlightServerManager.runFlashlight(FlashlightServerManager.java:160)\r\n\tat org.lantern.proxy.pt.FlashlightServerManager.startFlashlight(FlashlightServerManager.java:112)\r\n\tat org.lantern.proxy.pt.FlashlightServerManager.update(FlashlightServerManager.java:103)\r\n\tat org.lantern.proxy.pt.FlashlightServerManager.onModeChanged(FlashlightServerManager.java:76)\r\n\tat sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)\r\n\tat sun.reflect.NativeMethodAccessorImpl.invoke(Unknown Source)\r\n\tat sun.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source)\r\n\tat java.lang.reflect.Method.invoke(Unknown Source)\r\n\tat com.google.common.eventbus.EventSubscriber.handleEvent(EventSubscriber.java:74)\r\n\tat com.google.common.eventbus.SynchronizedEventSubscriber.handleEvent(SynchronizedEventSubscriber.java:47)\r\n\tat com.google.common.eventbus.EventBus.dispatch(EventBus.java:322)\r\n\tat com.google.common.eventbus.AsyncEventBus.access$001(AsyncEventBus.java:34)\r\n\tat com.google.common.eventbus.AsyncEventBus$1.run(AsyncEventBus.java:117)\r\n\tat java.util.concurrent.ThreadPoolExecutor.runWorker(Unknown Source)\r\n\tat java.util.concurrent.ThreadPoolExecutor$Worker.run(Unknown Source)\r\n\tat java.lang.Thread.run(Unknown Source)\r\nCaused by: java.io.FileNotFoundException: C:\Users\Julio\AppData\Roaming\Lantern\flashlight.exe (El proceso no tiene acceso al archivo porque est? siendo utilizado por otro proceso)\r\n\tat java.io.FileOutputStream.open(Native Method)\r\n\tat java.io.FileOutputStream.<init>(Unknown Source)\r\n\tat java.io.FileOutputStream.<init>(Unknown Source)\r\n\tat org.lantern.LanternUtils.extractFileFromJar(LanternUtils.java:1313)\r\n\tat org.lantern.LanternUtils.extractExecutableFromJar(LanternUtils.java:1259)\r\n\tat org.lantern.proxy.pt.BasePluggableTransport.<init>(BasePluggableTransport.java:64)\r\n\t... 17 more\r\n", "extra" : { "osArch" : "x86", "osVersion" : "6.2", "timeZone" : "America/Mexico_City", "version" : "1.5.10-21f3e4d", "country" : "MX", "fallback" : false, "threadName" : "Lantern-Async-Event-Thread-16", "logLevel" : "ERROR", "lineNumber" : 440, "javaVersion" : "1.7.0_25", "language" : "es", "methodName" : "uncaughtException", "osName" : "Windows 8" }, "nSimilarSuppressed" : 0, "key" : "org.littleshoot.proxy.impl.DefaultHttpProxyServer$ServerGroup$1.uncaughtException(DefaultHttpProxyServer.java:440)" }

--- a/src/main/java/org/lantern/LanternUtils.java
+++ b/src/main/java/org/lantern/LanternUtils.java
@@ -1259,7 +1259,7 @@ public class LanternUtils {
     public static File extractExecutableFromJar(final String path, 
             final File dir) throws IOException {
         final File newFile = extractFileFromJar(path);
-        File oldFile = new File(dir.getPath() + "/" + newFile.getName());
+        File oldFile = new File(dir, newFile.getName());
         if (oldFile.exists()) {
             HashCode newHash = Files.hash(newFile, Hashing.sha256());
             HashCode oldHash = Files.hash(newFile, Hashing.sha256());

